### DISCUSSION TITLE
Testimonial thread - how I'm using Script Kit

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#44] [General]
Hey John:

I don't remember how I bumped into simplescripts first - maybe twitter because you commented on a tweet by somebody I follow. I'm impressed by the structure of the this thing and the time you've put into sharing some of the development on twitch, and how responsive you've been to support questions.

My use cases aren't very sophisticated yet, I'm afraid. I've been learning javascript and have been writing a few small tools for myself in iOS/mac Scriptable or in Drafts, or in Google Sheets scripts, making my way through this year's Advent of Code challenges, etc. I'm still coming to terms with some of javascript's shorthands for declaring functions, restructuring data, etc. So much of what I do isn't necessarily helping me or anybody that much in a daily practical way as much as the coding is a means to learn coding. 

That said, the sessions on twitch that you led on getting/setting window parameters helped me dive into AppleScript more so that I feel more comfortable exploring new approaches with that now. I've got a little script based on that that launches five apps and places their windows around my monitor just so when I'm running Zoom sessions for an org I work with. 

I'm always wanting data from a CRM database we use. Simplescripts is so fast that I'm now happy just running a script that queries the database via API and gives me a menu is simple of info to copy in real time.

My mac has tools like Keyboard Maestro, BTT, Hazel etc, and I could probably build some of these things using those tools, but I like how readily I can configure the arrangement of how scripts are presented in the interface, how quickly you can make selections from the keyboard. Structuring the approach with simple scripts is less overwhelming than the many options some of those tools provide, plus I have a little satisfaction anytime I successfully debug my own script.

the set-three-column-windows script you posted a few days ago is one I use often. Selecting the windows to add via this list presented in real time in the app interface is smoother than other modes of doing this where you might mouse around to click on a window, run a macro to position it, repeat same two more times. 

I like that there are binary replicas of the script that can be called from the command line. I've got one hazel script that watches a cloud folder for a file, grabs its content, and passes that content directly to a simplesimplescript  via a shell command  - it lets me launch a process remotely when I'm away from the computer.

More experienced coders will surely make better use of the tool than I am now, but the mix of elegant interface, the ease of mixing AppleScript and javascript, and the pleasure of puzzling out my own solutions to things are making simplescripts a great experience for me.

thanks for all your work!

**Comment by johnlindquist** [comment]
Wow, thank you so much for sharing this. This means the world to me!

**Comment by siraj-samsudeen** [comment]
Hi John,
I was sold on the idea of ScriptKit when I watched your session on LearnWithJason. I am a false beginner to JS, neither a beginner, nor someone capable of doing stuff without having to think much. I was looking for a way to actively practice my JS skills as I know most of the tutorial content - I wanted a way to learn by doing - yours was a perfect invitation. 

I have started a new practice of solving problems on leetcode daily - I want to code in my local VSCode, not in leetcode. Hence, I copy the problem name, create the file manually, then copy the URL as a comment on the first line. 
Today I automated this in about 1 hour of trying - feeling really good. I wanted to do a few more things, but not able to. But this is so good for me to get started with the Kit - this made my day, John...Thank you so much...

**Reply by SimplGy** [reply]
(for others) Video URL: https://www.youtube.com/watch?v=zu_g17ePajA
